components/molecule/collapsible/src/index.js

```diff
@@ -75,16 +79,25 @@ const MoleculeCollapsible = ({
       </div>
       {showButton && (
         <div className={containerClassName}>
-          <button
-            type="button"
-            className={BUTTON_CLASS}
-            onClick={toggleCollapse}
-          >
-            <span className={BUTTON_CONTENT_CLASS} tabIndex="-1">
+          {ToggleButton ? (
+            <ToggleButton
+              onClick={toggleCollapse}
+              rightIcon={<span className={iconClassName}>{icon}</span>}
+            >
               {collapsed ? showText : hideText}
-              <span className={iconClassName}>{icon}</span>
-            </span>
-          </button>
+            </ToggleButton>
+          ) : (
+            <button
+              type="button"
+              className={BUTTON_CLASS}
+              onClick={toggleCollapse}
+            >
+              <span className={BUTTON_CONTENT_CLASS} tabIndex="-1">
+                {collapsed ? showText : hideText}
+                <span className={iconClassName}>{icon}</span>
+              </span>
+            </button>
```

Comment:
This part whould be componentized on a different file. Call it `CollapseToogleButton`.